File: src/main/java/com/msg/gauth/domain/auth/RefreshToken.java
package com.msg.gauth.domain.auth;

import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;
import org.springframework.data.redis.core.index.Indexed;

import javax.persistence.Id;

@RedisHash
public class RefreshToken {
    @Id
    Long userId;

    @Indexed
    String token;

    @TimeToLive
    Long timeToLive;

    public void updateToken(String token, Long timeToLive) {
        this.token = token;
        this.timeToLive = timeToLive;
    }
}

File: src/main/kotlin/com/msg/gauth/domain/auth/RefreshToken.kt
package com.msg.gauth.domain.auth

import com.msg.gauth.global.security.jwt.JwtTokenProvider
import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.index.Indexed

@RedisHash(value = "refreshToken", timeToLive = JwtTokenProvider.REFRESH_EXP)
class RefreshToken(
    @Id
    @Indexed
    val userId: Long,
    @Indexed
    val token: String,
)